### PR TITLE
start e2e testing for SDK/Tendermint clients

### PR DESCRIPTION
**Pull request by ilgooz**
This adds `clienttest` pkg to create a testing environment for SDK and Tendermint clients. As a start, I converted one SDK client test from unit to e2e and it seems that we already caught the first bug there! (please check the CI, tests won't pass because of an [unexpected response type- line 185](https://github.com/tendermint/cosmos-rosetta-gateway/runs/1161048609?check_suite_focus=true#step:4:185))

I'll also convert other unit tests in the clients to e2e if you guys also agree to convert. Otherwise we can do black-box testing in the `clienttest` (but need to rename as `client_test`) pkg if converting unit tests to e2e is not desired. But to me, converting most of them -if they test stuff related to API responses- will make the tests get closer to the truth.

### CODE REVIEW

**Review by faddat — COMMENTED** [review]
I take it this accomplishes the same thing as the build-test.yml thing I was working on?

**Comment by ilgooz** [comment]
I'm not sure. With this one, I'm serving a clean starport app -with no state from previous tests- for each _TestX_ func. Maybe your version starts serving a starport app as a GH service and keeps it running during the whole `go test ./...`? If that, I think use cases are different.

**Comment by faddat** [comment]
So what mine does is it generates a starter port app runs the starport app, starts CRG, and then executes queries against it.

The idea with mine was to test it every single time. Yours looks quite similar. I will put a little time into mine tonight and maybe we can compare notes later.

**Comment by ilgooz** [comment]
Some other goals:
- Make sure RPC servers are ready to accept connections
- Providing Launchpad/Stargate options at the same time
- Providing some test utility funcs

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.14'
+          go-version: '1.15.2'
```
Comment (alessio) [inline .github/workflows/test.yml:17]:
```suggestion
          go-version: '1.15'
```

**Comment by ilgooz** [comment]
Is this good to go @jgimeno @sahith-narahari @alessio?

**Comment by jgimeno** [comment]
The linter is failing @ilgooz

**Comment by ilgooz** [comment]
Fixed

**Review by sahith-narahari — APPROVED** [review]
lgtm

**Review by jgimeno — APPROVED** [review]

**Pull request merged**